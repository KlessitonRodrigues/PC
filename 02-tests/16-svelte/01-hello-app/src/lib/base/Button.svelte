<script>
  export let label = "Start";
</script>

<button class="bg-blue-500 text-white px-6 py-1 rounded-md hover:bg-blue-700"
  >{label}</button
>

<style>
</style>
