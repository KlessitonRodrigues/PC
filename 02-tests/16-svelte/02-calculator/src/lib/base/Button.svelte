<button class="button">
  <slot />
</button>

<style lang="scss">
  .button {
    background-color: $mainBg;
  }
</style>
