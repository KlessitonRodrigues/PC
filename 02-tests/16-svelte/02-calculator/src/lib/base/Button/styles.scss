.button {
  display: flex;
  align-items: center;
  padding: 0 size(10);
  min-height: size(12);
  background-color: $mainBg;
  color: $mainText;
  border-radius: $radiusSmall;
  position: relative;
  overflow: hidden;
  cursor: pointer;
  font-weight: bold;

  &::before {
    content: "";
    position: absolute;
    height: 100%;
    width: 0%;
    top: 0;
    left: -40px;
    transform: skewX(45deg);
    background-color: $main;
    transition: all 0.3s ease-out;
  }

  &:hover::before {
    width: 175%;
  }

  span {
    z-index: 1;
  }
}
