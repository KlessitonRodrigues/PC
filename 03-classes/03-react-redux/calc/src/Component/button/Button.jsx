import React from "react"

const Button = props => {

    return (
        <button className="calc-btn">
            {props.label}
        </button>
    )
}
export default Button 