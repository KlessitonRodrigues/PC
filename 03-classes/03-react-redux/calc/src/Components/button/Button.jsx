import React from "react"

let Button = props => {

    let btnType = ""

    if (props.label === "Del") {
        btnType = " btn-delete"
    } else if (props.label === "AC") {
        btnType = " btn-clear"
    } else if (/\d/.test(props.label)) {
        btnType = " btn-digit"
    } else if (/\D/.test(props.label)) {
        btnType = " btn-operator"
    }

    return (
        <button className={"calc-btn" + btnType}>
            {props.label}
        </button>
    )
}
export default Button 