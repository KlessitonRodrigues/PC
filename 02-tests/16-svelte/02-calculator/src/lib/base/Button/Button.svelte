<script lang="ts">
  import "./styles.scss";
  export let type: "submit" | "button" = "button";
  export let onClick = () => {};
</script>

<button {type} class="button" on:click={onClick}>
  <span>
    <slot />
  </span>
</button>
