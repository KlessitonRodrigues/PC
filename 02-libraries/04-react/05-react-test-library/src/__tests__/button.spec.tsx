import React from "react";
import renderer from "react-test-renderer";
import Button from "../button/index";

describe("button test", () => {
  it("Should render correctly ", () => {
    const component = renderer
      .create(<Button text="test" onClick={() => {}} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
