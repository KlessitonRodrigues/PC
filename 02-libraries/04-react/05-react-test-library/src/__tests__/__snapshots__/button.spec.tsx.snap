// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test Should render correctly  1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#888",
      "borderRadius": "4px",
      "fontFamily": "monospace",
      "fontSize": "20px",
      "fontWeight": "bold",
      "padding": "0 0.5rem",
    }
  }
>
  test
</button>
`;
